python/pyspark/sql/connect/plan.py

```diff
@@ -537,7 +537,7 @@ def __init__(
         self,
         left: Optional["LogicalPlan"],
         right: "LogicalPlan",
-        on: "ColumnOrString",
+        on: Optional[Union[str, List[str], ColumnRef]],
```

Comment:
Should actually use the signature of either `Sequence` or `Iterable[ColumnRef]` for `str` and `ColumnRef`. Should fix `dataframe.py` in PySpark API too. I am fine as is since it's at least consistent now.

Reply:
Is `list` to `Sequence` or `Iterable[xxx]` a API incompatible change? If not I can try to fix in this PR which sounds straightforward.

Reply:
should not be incompatible change, they are just used for static checking

Reply:
hmmm I gave a try but failed to make it work for mypy and UT together (not a python expert):

I found the trick case is `str` seems to be also consider-red as `Sequence` or `Iterable`.